### PR TITLE
Map terraforming danger ores april2019

### PR DESCRIPTION
**Pull request by TWLTriston**
Modifications to danger ores scenario, for peer review, not comprehensively play tested.

### CODE REVIEW

**Comment by TWLTriston** [comment]
Known issue - biters in unrevealed tiles still spawn post the "win" condition

File: map_gen/maps/terraforming_danger_ores.lua [hunk 1]
```diff
@@ -452,6 +556,59 @@ map = b.fish(map, 0.025)
...
+                'Biters at maximum evolution! Protect the base for an additional 100 rockets to wipe them out forever.'
+            game.print(message)
+        end
+
+        game.forces.enemy.evolution_factor = current_evolution
+        if win_condition_evolution_rocket_maxed > 0 and satellite_count >= (win_condition_evolution_rocket_maxed + 100) then
+            message = 'Congratulations! Biters have been wiped from the map!'
+            game.print(message)
+
+            win_condition_evolution_rocket_maxed = true
```
Comment (grilledham) [inline map_gen/maps/terraforming_danger_ores.lua]: Is this meant to be `win_condition_biters_disabled `? The variable changes from a number to a Boolean which seems odd.

**Comment by grilledham** [comment]
I fixed a problem with the random quadrants. Before if you saved and loaded, the game would crash saying you can't use `math.random` in the load event. If it didn't crash it would have caused a desync anyway.

`win_condition_evolution_rocket_maxed` and `win_condition_biters_disabled` need to be stored in global if you are going to change them at runtime, else the game will desync.
This change still needs to be made.

Otherwise, I don't see anything else that looks problematic.

File: map_gen/maps/terraforming_danger_ores.lua (hunk 1 above)
Comment (TWLTriston) [reply map_gen/maps/terraforming_danger_ores.lua]: Yes, thank you!

**Comment by grilledham** [comment]
We played this, as far as I can tell it worked, so let's merge.

**Pull request merged**